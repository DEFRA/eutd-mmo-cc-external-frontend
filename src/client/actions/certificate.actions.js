import { isError, showFullPageError } from './index';

export const GET_SUMMARY_DOCUMENT = 'GET_SUMMARY_DOCUMENT';
export const SHOW_SUMMARY_DOCUMENT_ERRORS = 'SHOW_SUMMARY_DOCUMENT_ERRORS';
export const GET_SUMMARY_UNAUTHORISED = 'GET_SUMMARY_DOCUMENT_UNAUTHORISED';

export function showSummaryCertificateErrors(data) {
  return { type: SHOW_SUMMARY_DOCUMENT_ERRORS, payload: { validationErrors: data } };
}

export const getSummaryCertificate = (journey, documentNumber = undefined) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? { headers: { documentnumber: documentNumber } }
    : {};

  await orchestrationApi.get(`/certificate/${journey}`, config)
    .then(
      res => {
        dispatch({ type: GET_SUMMARY_DOCUMENT, payload: res.data });
      },
      error => {
        if (isError(error)) {
          dispatch(showFullPageError());
        } else {
          dispatch(unauthorised());
        }
      });

    function unauthorised() { return { type: GET_SUMMARY_UNAUTHORISED }; }
};