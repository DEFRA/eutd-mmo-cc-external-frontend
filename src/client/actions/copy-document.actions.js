import { showFullPageError, apiCallFailed, isError, CLEAR_ERRORS } from './index';
export const CONFIRM_COPY_DOCUMENT            = 'confirm_copy_document';
export const UNAUTHORISED_COPY_DOCUMENT       = 'unauthorised_copy_document';
export const CLEAR_COPY_DOCUMENT              = 'clear_copy_document';
export const REMOVE_CONFIRM_COPY_CERTIFICATE  = 'remove_confirm_copy_certificate';
export const ADD_ERROR_COPY_CERTIFICATE       = 'add_error_copy_certificate';
export const REMOVE_ERROR_COPY_CERTIFICATE    = 'remove_error_copy_certificate';

export const submitCopyCertificate = (payload) => async (dispatch, _getState, {orchestrationApi}) => {
  const {documentNumber, copyDocumentAcknowledged, voidOriginal, excludeLandings, journey} = payload;
  const config = {headers: {documentnumber: documentNumber}};

  try {
    let ipAddress;
    if (voidOriginal) {
      const ip = await orchestrationApi.get('/client-ip');
      ipAddress = ip.data;
    }

    return await orchestrationApi.post('/confirm-copy-certificate', { copyDocumentAcknowledged, voidOriginal, excludeLandings, journey, ipAddress }, config)
    .then(res => {
      dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: {
        documentNumber: res.data.documentNumber,
        copyDocumentAcknowledged: res.data.copyDocumentAcknowledged,
        voidDocumentConfirm: res.data.voidOriginal
      }});
      return res.data.newDocumentNumber;
    });
  } catch (e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch({ type: UNAUTHORISED_COPY_DOCUMENT }); return;
    } else {
      dispatch(apiCallFailed(e.response));
    }
  }
};

export const checkCopyCertificate = (documentNumber) => async (dispatch, _getState, {orchestrationApi}) => {
  const config = {
    headers: {
      documentnumber: documentNumber
    }
  };

  try {
    const res = await orchestrationApi.get('/check-copy-certificate', config);
    const { canCopy } = res.data;
    if (!canCopy) {
      dispatch(unauthorised());
    }
  } catch(e) {
    if (isError(e)) dispatch(showFullPageError());
    if (e.response && e.response.status === 403) {
      dispatch(unauthorised());
      throw new Error(`Unauthorised access ${e.response.status}`);
    } else {
      dispatch(apiCallFailed(e.response));
      throw new Error(`An error has occurred ${e.response.status} (${e.response.statusText})`);
    }
  }

  function unauthorised() { return { type: UNAUTHORISED_COPY_DOCUMENT }; }
};

export const addCopyDocument = (data) => (dispatch) => dispatch({ type: CONFIRM_COPY_DOCUMENT, payload: data });
export const clearErrors = () => (dispatch) =>  dispatch({ type: CLEAR_ERRORS });
export const clearCopyDocument = () => (dispatch) => dispatch({ type: CLEAR_COPY_DOCUMENT });
export const unauthorisedCopyDocument = () => (dispatch) => dispatch({ type: UNAUTHORISED_COPY_DOCUMENT });