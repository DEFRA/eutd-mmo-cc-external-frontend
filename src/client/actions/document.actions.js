// actions
import {isError, showFullPageError} from './index';

export const GOT_ALL_DOCUMENTS          = 'GOT_ALL_DOCUMENTS';
export const GET_ALL_DOCUMENTS_FAILED   = 'GET_ALL_DOCUMENTS_FAILED';
export const GET_COMPLETED_DOCUMENT = 'GET_COMPLETED_DOCUMENT';
export const GET_PENDING_DOCUMENT = 'GET_PENDING_DOCUMENT';


// action creators
export const getAllDocuments = (type, month, year) => async (dispatch, getState, { orchestrationApi }) => {
  try {
    const res = await orchestrationApi.get(`/documents/${year}/${month}?type=${type}`);
    return dispatch({ type: GOT_ALL_DOCUMENTS, payload: res.data });

  } catch(e) {
    if( isError(e) ) dispatch(showFullPageError());
    //throw new Error(e.response);
  }
};

export const getCompletedDocument = (documentNumber) => async (dispatch, getState, { orchestrationApi }) => {
  try {
    const res = await orchestrationApi.get(`/document/${documentNumber}`);
    if (res.data && res.data.documentStatus === 'COMPLETE') {
      return dispatch({ type: GET_COMPLETED_DOCUMENT, payload: res.data });
    }
    return dispatch({ type: GET_COMPLETED_DOCUMENT, payload: null });
  }
  catch (e) {
    return dispatch({ type: GET_COMPLETED_DOCUMENT, payload: null });
  }
};

export const getPendingDocument = (documentNumber) => async (dispatch, getState, { orchestrationApi }) => {
  try {
    const res = await orchestrationApi.get(`/document/${documentNumber}`);
    if (res.data && res.data.documentStatus === 'PENDING') {
      return dispatch({ type: GET_PENDING_DOCUMENT, payload: res.data });
    }
    return dispatch({ type: GET_PENDING_DOCUMENT, payload: null });
  }
  catch (e) {
    return dispatch({ type: GET_PENDING_DOCUMENT, payload: null });
  }
};