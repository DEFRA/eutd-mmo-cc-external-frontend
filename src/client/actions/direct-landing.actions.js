import { beginApiCall, apiCallFailed, isForbidden, showFullPageError, CLEAR_ERRORS, isSilverLineError } from './index';
import { exportPayloadActionTypes } from '../actions';


export const GET_DIRECT_LANDING_PRODUCTS = 'get_direct_landing_products';
export const CLEAR_DIRECT_LANDING = 'clear_direct_landing';
export const DIRECT_LANDING_PRODUCTS_FAILED = 'direct_landing_products_failed';
export const POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'post_direct_landing_products_unauthorised';
export const GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'get_direct_landing_products_unauthorised';
export const POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR = 'post_direct_landing_prodcts_waf_error';

export const getDirectLandings =
  (documentNumber) =>
  async (dispatch, _getState, { orchestrationApi }) => {
    const config = { headers: { documentnumber: documentNumber } };

    try {
      dispatch(beginApiCall());

      const res = await orchestrationApi.get(
        '/export-certificates/export-payload/direct-landings',
        config
      );

      await dispatch({ type: GET_DIRECT_LANDING_PRODUCTS, payload: res.data });
    } catch (e) {
      if (isForbidden(e)) {
        dispatch({ type: GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
      } else {
        dispatch(apiCallFailed(e.response));
        throw new Error(
          `An error has occurred ${e.response.status} (${e.response.statusText})`
        );
      }
    }
  };

export const clearDirectLanding = () => ({ type: CLEAR_DIRECT_LANDING });

export const upsertDirectLanding = (landing, documentNumber = undefined) => (dispatch, getState, { orchestrationApi } ) => {
  const config = (documentNumber)
    ? {
        headers: {
          documentnumber: documentNumber
        }
    }
    : {};

  return orchestrationApi.post('/export-certificates/direct-landing/validate', landing, config)
    .then(
      res => {
        dispatch(landingUpserted(res.data));
        dispatch(clearErrors());
      },
      error => {
        if (error.response.status === 400) {
          dispatch(landingUpserted(error.response.data));
          throw new Error();
        } else if (isSilverLineError(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR, supportID: error.response.data.match(/\d+/)[0] });
        } else if (isForbidden(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
        } else {
          dispatch(showFullPageError());
        }
      });

  function landingUpserted(exportPayload) { return { type: exportPayloadActionTypes.LANDING_UPSERTED, exportPayload, errors: exportPayload.errors }; }
  function clearErrors() { return { type: CLEAR_ERRORS }; }
};
