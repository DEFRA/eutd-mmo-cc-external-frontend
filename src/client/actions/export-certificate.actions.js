import { showSummaryCertificateErrors } from './certificate.actions';

export const exportCertificateActionTypes = {
  EXPORT_CERT_CREATE_REQUEST: 'export-certificate/create',
  EXPORT_CERT_CREATE_REQUEST_SUCCESS: 'export-certificate/create/success',
  EXPORT_CERT_CREATE_REQUEST_FAILED: 'export-certificate/create/failed'
};

export const createExportCertificate = (currentUri, journey, noOfVessels, documentNumber) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? {headers: {documentNumber: documentNumber}}
    : {};

  const { data } = await orchestrationApi.get('/client-ip');

  return await orchestrationApi.post('/export-certificates/create', { journey, data },config)
    .then(
      res => {
        let certDetails = res.data;
        if (certDetails.status && certDetails.status === 'invalid catch certificate') {
          return 'ERROR';
        }
        window.dataLayer = window.dataLayer || [];
        window.dataLayer.push({
          'event': 'completedCatchCertificate',
          'numberOfVessels': noOfVessels
        });
        return dispatch(exportCertificateCreated(certDetails));
      },
      error => {
        dispatch(exportCertificateCreateFailed(error.toString()));
        if(error.response && error.response.data) {
          return dispatch(showSummaryCertificateErrors(error.response.data));
        }
      })
    .catch((err) => {
      dispatch(exportCertificateCreateFailed(err));
      throw(err);
    });

  function exportCertificateCreated(certDetails) { return { type: exportCertificateActionTypes.EXPORT_CERT_CREATE_REQUEST_SUCCESS,  payload: { data: certDetails } }; }
  function exportCertificateCreateFailed(errorRes) { return { type: exportCertificateActionTypes.EXPORT_CERT_CREATE_REQUEST_FAILED, errorRes }; }
};

export const getExportCertificateFromParams = (params) => (dispatch) => {
  return dispatch({type: exportCertificateActionTypes.EXPORT_CERT_CREATE_REQUEST_SUCCESS, payload: {data: params }});
};