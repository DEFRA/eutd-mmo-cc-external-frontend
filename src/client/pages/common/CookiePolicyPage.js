import React from 'react';
import { getAllUserAttributes } from '../../actions/userAttributes.actions';
import CookiePolicyWelshPage from './CookiePolicyWelshPage';
import CookiePolicyEnglishPage from './CookiePolicyEnglishPage';
import { connect } from 'react-redux';
import PageTemplateWrapper from '../../components/PageTemplateWrapper';

class CookiePolicyPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: 'No',
      showSuccessBanner: false,
    };
  }

  handleSaveCookieSettingClick = () => {
    if (this.state.selected === 'Yes') {
      document.cookie =
        'analytics_cookies_accepted=true;expires=Thu, 31 Dec 2099 23:59:59 UTC;samesite;secure';
    } else {
      document.cookie =
        'analytics_cookies_accepted=false;expires=Thu, 31 Dec 2099 23:59:59 UTC;samesite;secure';
    }
    this.setState({ showSuccessBanner: true });
    window.scrollTo(0, 0);
  };

  handleChange = (option) => {
    this.setState({ selected: option });
  };

  componentDidMount() {
    const cookieValue =
      document.cookie &&
      document.cookie
        .split(';')
        .some((item) => item.includes('analytics_cookies_accepted'));
    cookieValue === true
      ? this.setState({ selected: 'Yes' })
      : this.setState({ selected: 'No' });
  }

  getSelectedLanguage = (userAttributes) => {
    const userLanguageAttribute = userAttributes.find(
      (userAttribute) => userAttribute.name === 'language'
    );
    if (userLanguageAttribute) {
      return userLanguageAttribute.value;
    }
  };

  render() {
    const { userAttributes } = this.props;
    const userSelectedLanguage = this.getSelectedLanguage(userAttributes);

    return userSelectedLanguage === 'cy_UK' ? (
      <CookiePolicyWelshPage
        showSuccessBanner={this.state.showSuccessBanner}
        saveCookieSetting={this.handleSaveCookieSettingClick}
        selected={this.state.selected}
        onChange={this.handleChange}
      />
    ) : (
      <CookiePolicyEnglishPage
        showSuccessBanner={this.state.showSuccessBanner}
        saveCookieSetting={this.handleSaveCookieSettingClick}
        selected={this.state.selected}
        onChange={this.handleChange}
      />
    );
  }
}

function mapStateToProps(state) {
  return {
    userAttributes: state.userAttributes,
  };
}

function loadData(store) {
  return Promise.all([store.dispatch(getAllUserAttributes())]);
}

export const component = connect(mapStateToProps, {
  getAllUserAttributes,
})(CookiePolicyPage);

export default {
  loadData,
  component: PageTemplateWrapper(component),
};