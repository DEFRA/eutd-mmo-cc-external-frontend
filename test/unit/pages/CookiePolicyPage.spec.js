import React from 'react';
import { mount } from 'enzyme';
import { component as CookiePolicyPage } from '../../../src/client/pages/common/CookiePolicyPage';
import CookiePolicyWholePage from '../../../src/client/pages/common/CookiePolicyPage';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { getAllUserAttributes } from '../../../src/client/actions/userAttributes.actions';
jest.mock('../../../src/client/actions/userAttributes.actions');

let wrapper;
const mockStore = configureStore([thunk]);
const goBack = jest.fn();

const getWrapper = (language) => {
  const store = mockStore({
    userAttributes: [
      {
        name: 'language',
        value: language,
        modifiedAt: '2022-03-15T21:47:10.755Z',
      },
    ],
  });
  const props = {
    window: {
      history: {
        goBack: goBack(),
      },
    },
  };
  wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <CookiePolicyPage {...props} />
      </MemoryRouter>
    </Provider>
  );

  return wrapper;
};

describe('Cookie Policy page in English or welsh', () => {
  it('should load successfully in English or welsh', () => {
    const englishWrapper = getWrapper('en_UK');
    const welshWrapper = getWrapper('cy_UK');
    expect(englishWrapper).toBeDefined();
    expect(welshWrapper).toBeDefined();
  });

  it('should render save cookies settings button in English or welsh', () => {
    const englishWrapper = getWrapper('en_UK');
    const welshWrapper = getWrapper('cy_UK');
    expect(englishWrapper.find('#saveCookieSettings')).toBeDefined();
    expect(englishWrapper.find('#saveCookieSettings').text()).toEqual(
      'Save cookie settings'
    );
    expect(welshWrapper.find('#saveCookieSettings')).toBeDefined();
    expect(welshWrapper.find('#saveCookieSettings').text()).toEqual(
      'Cadw’r gosodiadau cwcis'
    );
  });

  it('should render radio button in English or welsh', () => {
    const englishWrapper = getWrapper('en_UK');
    const welshWrapper = getWrapper('cy_UK');
    expect(englishWrapper.find({ type: 'radio' })).toBeDefined();
    expect(welshWrapper.find({ type: 'radio' })).toBeDefined();
  });

  it('pushes back a page when save cookies settings button in English or welsh is clicked', () => {
    const englishWrapper = getWrapper('en_UK');
    const welshWrapper = getWrapper('cy_UK');
    englishWrapper.find('#saveCookieSettings').simulate('click');
    welshWrapper.find('#saveCookieSettings').simulate('click');
    expect(goBack).toHaveBeenCalled();
  });

  it('should handle onChange event on Yes option in English or welsh', () => {
    const englishWrapper = getWrapper('en_UK');
    const welshWrapper = getWrapper('cy_UK');
    englishWrapper.setState({
      selected: 'Yes',
    });
    welshWrapper.setState({
      selected: 'Yes',
    });
    expect(
      englishWrapper
        .find('input#cookieAnalyticsAccept')
        .simulate('change', { preventDefault() {} })
    ).toBeTruthy();
    expect(
      welshWrapper
        .find('input#cookieAnalyticsAccept')
        .simulate('change', { preventDefault() {} })
    ).toBeTruthy();
  });

  it('should handle onChange event on No option in English or welsh', () => {
    const englishWrapper = getWrapper('en_UK');
    const welshWrapper = getWrapper('cy_UK');

    englishWrapper.setState({
      selected: 'No',
    });
    expect(
      englishWrapper
        .find('input#cookieAnalyticsReject')
        .simulate('change', { preventDefault() {} })
    ).toBeTruthy();

    welshWrapper.setState({
      selected: 'No',
    });
    expect(
      welshWrapper
        .find('input#cookieAnalyticsReject')
        .simulate('change', { preventDefault() {} })
    ).toBeTruthy();
  });

  it('should handle onClick event on the Save cookie settings in English or welsh', () => {
    const englishWrapper = getWrapper('en_UK');
    const welshWrapper = getWrapper('cy_UK');
    expect(
      englishWrapper
        .find('#saveCookieSettings')
        .simulate('click', { preventDefault() {} })
    ).toBeTruthy();
    expect(
      welshWrapper
        .find('#saveCookieSettings')
        .simulate('click', { preventDefault() {} })
    ).toBeTruthy();
  });

  it('should handle onClick event on the Go back to the page you were looking at link in English or welsh', () => {
    const englishWrapper = getWrapper('en_UK');
    const welshWrapper = getWrapper('cy_UK');

    englishWrapper.find('#saveCookieSettings').simulate('click', {});
    englishWrapper.find('a.notification-banner__link').simulate('click', {});

    expect(goBack).toHaveBeenCalled();

    welshWrapper.find('#saveCookieSettings').simulate('click', {});
    welshWrapper.find('a.notification-banner__link').simulate('click', {});

    expect(goBack).toHaveBeenCalled();
  });

  it('should display notification banner with setting cookie preferences in English or welsh', () => {
    const englishWrapper = getWrapper('en_UK');
    const welshWrapper = getWrapper('cy_UK');

    englishWrapper.find('#saveCookieSettings').simulate('click', {});

    expect(englishWrapper.find('h2.notification-banner__title').text()).toEqual(
      'Success'
    );
    expect(
      englishWrapper.find('p.notification-banner__heading').text()
    ).toEqual(
      'You’ve set your cookie preferences.Go back to the page you were looking at.'
    );
    expect(englishWrapper.find('a.notification-banner__link')).toBeTruthy();

    welshWrapper.find('#saveCookieSettings').simulate('click', {});

    expect(welshWrapper.find('h2.notification-banner__title').text()).toEqual(
      'Llwyddiant'
    );
    expect(welshWrapper.find('p.notification-banner__heading').text()).toEqual(
      'Rydych chi wedi gosod eich dewisiadau cwcis.Mynd yn ôl i’r dudalen roeddech chi’n edrych arni.'
    );
    expect(welshWrapper.find('a.notification-banner__link')).toBeTruthy();
  });

  it('should take the snapshot of the page in English', () => {
    const wrapper = getWrapper('en_UK');
    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  });
});

describe('CookiePolicyPage, loadData', () => {
  const store = { dispatch: jest.fn() };

  beforeEach(() => {
    getAllUserAttributes.mockReset();
    getAllUserAttributes.mockReturnValue({ type: 'GET_ALL_USER_ATTRIBUTES' });
  });

  it('should call the getAllUserAttributes to load the component', () => {
    CookiePolicyWholePage.loadData(store);

    expect(getAllUserAttributes).toHaveBeenCalledTimes(1);
    expect(getAllUserAttributes).toHaveBeenCalled();
  });
});