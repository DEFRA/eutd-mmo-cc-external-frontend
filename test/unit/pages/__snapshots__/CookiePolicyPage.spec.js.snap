// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie Policy page in English or welsh should take the snapshot of the page in English 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      />
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="cookies_policy_header"
          >
            Cookies Policy
          </h1>
          <p>
            Cookies are small text files that are placed on your phone, tablet or computer by websites that you visit. They are widely used in order to make websites work, or work more efficiently, as well as to provide information to the owners of the site.
          </p>
          <p>
            The Fish Exports service uses cookies to store information about how you use our website, such as the pages you visit. The information below explains the cookies we use and why; e.g. Cookie Preferences and Google Analytics etc.
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="cookie_preference_header"
          >
            Cookie Preferences
          </h2>
          <p>
            This cookie is used to remember your choice about cookies. Where you have previously indicated a preference, your preference will be stored in this cookie.
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="google_analytics_header"
          >
            Google Analytics
          </h2>
          <p>
            These cookies are used to collect information about how you use our website. We use the information to compile reports and to help us improve the website. The cookies collect information such as the number of visitors to the website, where visitors have come to the website from and the pages they visited. This information does not identify anyone.
          </p>
          <p>
            We do not allow Google to use or share the data about how you use this site.
          </p>
          <p>
            Google Analytics sets cookies that store anonymised information about:
          </p>
          <ul
            class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              how you got to the site
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              the pages you visit on The Fish Exports services, and how long you spend on each page
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              how many documents and/or certificates are created
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              how you use the service so we can improve it
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              how we measure the performance of the digital service and track user journeys
            </li>
          </ul>
          <br />
          <p>
            Google is not allowed to use or share our analytics data, and you can opt out of Google Analytics  
            <a
              data-testid="here__link"
              href="https://tools.google.com/dlpage/gaoptout"
              rel="noopener noreferrer"
            >
              Here
            </a>
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="introductory_cookie_message_header"
          >
            Introductory Cookie Message
          </h2>
          <p>
            When you first use the service we show a ‘cookie message’. We then store a cookie on your phone, tablet or computer so it knows not to show this message again. This cookie will expire after one month, which means you’ll be asked the same question after this time.
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="cookies_that_remember_your_settings_header"
          >
            Cookies that remember your settings
          </h2>
          <p>
            These cookies remember your preferences and the choices you make, to personalise your experience when using the site.
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="essential_cookies_header"
          >
            Essential Cookies and Cookies you can choose
          </h2>
          <p>
            Different types of cookies do different jobs on our website. Some are needed to make the website work. We need your consent to use other cookies that are not essential. You can change your choices at any time. Just click the 'Cookies' link at the end of any page.
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="strictly_necessary_cookies_header"
          >
            Strictly Necessary Cookies
          </h2>
          <p>
            These essential cookies remember your progress through a form, for example a licence application, and always need to be on.
          </p>
          <p>
            If you are a customer, they help us know who you are so that you can sign in and manage your account. They also help us keep your details safe and private.
          </p>
          <p>
            Other important jobs they do are:
          </p>
          <ul
            class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Help you move around the site
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Tell us if you’ve been to it before and which pages you went to
            </li>
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Tell us how the site is working, so that we can find and fix any problems.
            </li>
          </ul>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="functional_cookies_header"
          >
            Functional Cookies
          </h2>
          <p>
            These cookies always need to be on and are used for remembering things like:
          </p>
          <ul
            class="ul e1vcssft0 css-t0nsp0-StyledList-StyledHoc e4u1u2g0"
          >
            <li
              class="e1vcssft0 css-h9lpae-StyledListItem-StyledHoc e4u1u2g0"
            >
              Your username and password on the page where you sign in
            </li>
          </ul>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="session_cookie_header"
          >
            Session cookie
          </h2>
          <p>
            We store session cookies on your phone, tablet or computer to help keep your information secure while you use the service. These cookies are automatically removed after 15 minutes following no activity or when you close your web browser.
          </p>
          <br />
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="change_cookie_settings_header"
          >
            Change your cookie settings
          </h2>
          <br />
          <h4
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
            data-testid="do_You_want_accept_analytics_cookies_header"
          >
            Do you want to accept the analytics cookies?
          </h4>
          <div
            id="radioButtons"
          >
            <fieldset
              class="govuk-fieldset"
            >
              <label
                class="css-sdrcg6-StyledHoc"
                for="cookieAnalyticsAccept"
                id="label-cookieAnalyticsAccept"
              >
                <input
                  class="css-1xt3i3l"
                  data-testid="cookieAnalyticsAccept"
                  id="cookieAnalyticsAccept"
                  name="cookieAnalyticsAccept"
                  type="radio"
                  value="Yes"
                />
                <span
                  class="css-1uwxoxa"
                >
                  Yes
                </span>
              </label>
              <label
                class="css-sdrcg6-StyledHoc"
                for="cookieAnalyticsReject"
                id="label-cookieAnalyticsReject"
              >
                <input
                  checked=""
                  class="css-1xt3i3l"
                  data-testid="cookieAnalyticsReject"
                  id="cookieAnalyticsReject"
                  name="cookieAnalyticsReject"
                  type="radio"
                  value="No"
                />
                <span
                  class="css-1uwxoxa"
                >
                  No
                </span>
              </label>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <button
            class="button"
            data-testid="saveCookieSettings"
            id="saveCookieSettings"
            type="button"
          >
            Save cookie settings
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
